proof/courier.go

```diff
@@ -159,6 +167,70 @@ func NewHashMailCourierAddr(addr url.URL) (*HashMailCourierAddr, error) {
 	}, nil
 }
 
+// UniverseRpcCourierAddr is a universe RPC protocol specific implementation of
+// the CourierAddr interface.
+type UniverseRpcCourierAddr struct {
+	addr url.URL
+}
+
+// Url returns the url.URL representation of the courier address.
+func (h *UniverseRpcCourierAddr) Url() *url.URL {
+	return &h.addr
+}
+
+// NewCourier generates a new courier service handle.
+func (h *UniverseRpcCourierAddr) NewCourier(_ context.Context,
+	cfg *CourierCfg, recipient Recipient) (Courier, error) {
+
+	// Ensure that the courier address is a universe RPC address.
+	if h.addr.Scheme != UniverseRpcCourierType {
+		return nil, fmt.Errorf("unsupported courier protocol: %v",
+			h.addr.Scheme)
+	}
+
+	// Connect to the universe RPC server.
+	dialOpts, err := serverDialOpts()
+	if err != nil {
+		return nil, err
+	}
+
+	serverAddr := fmt.Sprintf(
+		"%s:%s", h.addr.Hostname(), h.addr.Port(),
+	)
+	conn, err := grpc.Dial(serverAddr, dialOpts...)
+	if err != nil {
+		return nil, err
+	}
+
+	client := unirpc.NewUniverseClient(conn)
+
+	// Instantiate the events subscribers map.
+	subscribers := make(
+		map[uint64]*fn.EventReceiver[fn.Event],
```

Comment:
Thought there was some other higher level wrapper here?

Reply:
At the moment each courier type handles its own list of subscribers. I think it does make sense to add a higher level wrapper but we haven't done that yet and it's not in this PR.